managed/services/management/node.go

```diff
@@ -136,11 +137,37 @@ func (s *NodeService) Register(ctx context.Context, req *managementpb.RegisterNo
 		return nil, e
 	}
 
-	apiKeyName := fmt.Sprintf("pmm-agent-%s-%d", req.NodeName, rand.Int63()) //nolint:gosec
-	_, res.Token, e = s.akp.CreateAdminAPIKey(ctx, apiKeyName)
-	if e != nil {
-		return nil, e
+	if req.ExistedServiceToken == "" {
+		serviceAcountID, e := s.ap.CreateServiceAccount(ctx)
+		if e != nil {
+			return nil, e
+		}
+		_, res.Token, e = s.ap.CreateServiceToken(ctx, serviceAcountID)
+		if e != nil {
+			return nil, e
+		}
+	} else {
+		res.Token = req.ExistedServiceToken
 	}
 
 	return res, nil
 }
+
+// Unregister do unregistration of the node.
+func (s *NodeService) Unregister(ctx context.Context, req *managementpb.UnregisterNodeRequest) (*managementpb.UnregisterNodeResponse, error) {
+	err := s.db.InTransaction(func(tx *reform.TX) error {
+		return models.RemoveNode(tx.Querier, req.NodeId, models.RemoveCascade)
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	warning, err := s.ap.DeleteServiceAccount(ctx)
+	if err != nil {
+		return nil, err
```

Comment:
let's not fail in case we can't delete service account and just return a warning

Reply:
Done.